src/key/ed25519.ts

```diff
@@ -0,0 +1,56 @@
+import * as _ from "lodash";
+import { toArray, toHex } from "../utility";
+
+export type Ed25519Signature = string;
+
+export class Signature {
+    public sodium: any;
+
+    constructor() {
+        this.sodium = require("libsodium-wrappers-sumo");
+    }
+
+    public async init() {
+        await this.sodium.ready;
+    }
+
+    public signEd25519(message: string, priv: string): string {
+        return toHex(
+            this.sodium.crypto_sign_detached(toArray(message), toArray(priv))
+        );
+    }
+
+    public verifyEd25519 = (
+        signature: Ed25519Signature,
+        message: string,
+        pub: string
+    ): boolean => {
+        return this.sodium.crypto_sign_verify_detached(
+            toArray(signature),
+            toArray(message),
+            toArray(pub)
+        );
+    };
+}
+
+/// test
```

Comment:
You'd better to remove this test code or you can add it to test directory if needed.